x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -295,6 +340,20 @@ public List<String> argDescriptions() {
         }
     }
 
+    private static DataType getTargetType(String[] names) {
```

Comment:
Could you name this `autoStringLiteralConversion` or something? So we know it's about the auto conversions from string literals. It's not a generic "target" - it's a target for string literals.